stratum/hal/lib/barefoot/bf_sde_wrapper.cc

```diff
@@ -1344,13 +1344,25 @@ ::util::Status BfSdeWrapper::ConfigureQos(
   }
   device_to_ppg_handles_[device].clear();
   for (const auto& ppg_config : qos_config.ppg_configs()) {
+    uint32 sdk_port;
+    switch (ppg_config.port_type_case()) {
+      case TofinoConfig::TofinoQosConfig::PpgConfig::PortTypeCase::kSdkPort:
+        sdk_port = ppg_config.sdk_port();
+        break;
+      case TofinoConfig::TofinoQosConfig::PpgConfig::PortTypeCase::kPort: {
+        ASSIGN_OR_RETURN(sdk_port, GetPortIdFromPortKey(
+                                       device, PortKey(0, ppg_config.port())));
```

Comment:
Correct me if I am wrong
`PortKey(0, ppg_config.port())` will create a port key with -1 as channel value and `GetPortIdFromPortKey` will return failed status.

Reply:
Same for line 1392

Reply:
Could be, will test later.